pkg/debugapi/debugapi.go

```diff
@@ -125,6 +126,9 @@ func (s *Service) Configure(overlay swarm.Address, p2p p2p.DebugService, pingpon
 	s.chequebook = chequebook
 	s.swapEnabled = swapEnabled
 	s.swap = swap
+	if !chainEnabled {
```

Comment:
This extra parameter can be avoided by either initialising swap to `noOpSwap` before passing here or maybe checking if the `swap == nil`.